src/lib/common/sol-bus.c

```diff
@@ -383,9 +383,8 @@ sol_bus_client_free(struct sol_bus_client *client)
     if (!client)
         return;
 
-    destroy_client(client);
-
     sol_ptr_vector_remove(&_ctx.clients, client);
+    destroy_client(client);
```

Comment:
makes no sense to me since nobody changes the numeric value, it's basically an "unsigned long int".